File: Java/프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for (int i : tangerine) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }

        List<Integer> list = new ArrayList<>(map.keySet());
        list.sort((a, b) -> map.get(b) - map.get(a));

        for (Integer i : list) {
            if (k <= 0) {
                break;
            }

            answer++;
            k -= map.get(i);
        }

        return answer;
    }
}

File: Python/프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    d = {}

    for i in tangerine:
        d[i] = d.get(i, 0) + 1

    lst = list(d.keys())
    lst.sort(key=lambda i: d[i], reverse=True)

    for i in lst:
        if k <= 0:
            break

        answer += 1
        k -= d[i]

    return answer